lib/gears/dpi.lua

```diff
@@ -7,8 +7,87 @@
 -- @module gears.dpi
 ---------------------------------------------------------------------------
 
+local mround = require("gears.math").round
+
 local gdpi = {}
 
+local gdpi_mt = {}
+
+-- DPI rounding policy
+--
+-- Most monitors don't have a DPI which is a nice exact multiple of the
+-- reference pixel density (96). In such cases, users' preferences vary:
+-- some prefer using the DPI values as-is, resulting in fractional scaling,
+-- others prefer rounding the scaling factor (so that e.g. anything above 144
+-- DPI and below 240 should result in a scaling factor of 2).
+-- While this could be achieved by the users by manually forcing specific DPI
+-- values for the screens, it's rather annoying in dynamic configurations.
+-- So we offer the possibility for the user to choose a global policy,
+-- and all DPI usage will take this into consideration.
+-- Note that rounding is only applied to autocomputed DPI values, never
+-- when the user sets the DPI themselves.
```

Comment:
I can't say I'm convinced that this is a good idea, but at least the default is `identity`.
@Elv13 What do you think?